src/settings.rs

```diff
@@ -0,0 +1,74 @@
+use config::{Config, ConfigError, Environment, File};
+use serde::Deserialize;
+
+use log::LevelFilter;
+use url::Url;
+
+use std::{
+	collections::HashMap, env, net::SocketAddr, path::PathBuf, time::Duration,
+};
+
+#[derive(Debug, Deserialize)]
+pub struct BehaviorSettings {
+	pub max_keywords: u32,
+	patience_seconds: u64,
+}
+impl BehaviorSettings {
+	pub fn patience(&self) -> Duration {
+		Duration::from_secs(self.patience_seconds)
+	}
+}
```

Comment:
Can this use [`#[deserialize_with = "..."]`](https://serde.rs/field-attrs.html#deserialize_with) instead, so the field can just be a `Duration`?